internal/provider/resource_application.go

```diff
@@ -164,6 +165,34 @@ func (r *applicationResource) Schema(_ context.Context, _ resource.SchemaRequest
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
+			"storage": schema.SetNestedAttribute{
+				Description: "Configure storage constraints for the juju application.",
+				Optional:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"label": schema.StringAttribute{
+							Description: "The specific storage option defined in the charm.",
+							Required:    true,
+						},
+						"size": schema.StringAttribute{
+							Description: "The size of each volume. E.g. 100G.",
+							Required:    true,
+						},
+						"pool": schema.StringAttribute{
+							Description: "Name of the storage pool to use. E.g. ebs on aws.",
+							Optional:    true,
+						},
+						"count": schema.Int64Attribute{
+							Description: "The number of volumes.",
+							Optional:    true,
+						},
+					},
```

Comment:
We should do a plan validator similar to `setNestedIsAttributeUniqueValidator` for the objects. Ensure that we have one entry per label only. We may be able to reuse it too.

Reply:
thx. done